packages/api/src/providers/fitbit.ts

```diff
@@ -363,74 +380,57 @@ export class Fitbit extends Provider implements OAuth2 {
+      ProviderSource.fitbit,
+      fitbitUserId
+    );
+    const rejected: { userId: string; cxId: string; err: unknown }[] = [];
+    await Promise.allSettled(
+      connectedUsers.map(async user => {
+        if (user.dataValues.id !== currentUser.dataValues.id) {
+          try {
+            const token = getProviderTokenFromConnectedUserOrFail(user, ProviderSource.fitbit);
+            const fitbitToken = parseFitbitToken(token);
+            await this.oauth.revokeLocal(user);
+            const updatedUser = await getConnectedUserOrFail({
+              id: user.id,
+              cxId: user.cxId,
+            });
+            // intentionally asynchronous, not waiting for the result
+            sendProviderDisconnected(updatedUser, [ProviderSource.fitbit]);
+            const activeSubscriptions = await this.getActiveSubscriptions(fitbitToken);
+            if (activeSubscriptions.length)
+              await this.deleteSubscriptions(activeSubscriptions, fitbitToken.accessToken);
+            // eslint-disable-next-line @typescript-eslint/no-explicit-any
+          } catch (err: any) {
+            rejected.push({ userId: user.id, cxId: user.cxId, err: err.cause });
+          }
         }
       })
     );
-  }
 
-  // Finds existing users connected to this Fitbit account, revokes their tokens and sends WH updates about the disconnections
-  async revokeTokenFromOtherUsers(
-    currentUser: ConnectedUser,
-    fitbitUserId: string,
-    activeSubscriptions: FitbitWebhookSubscriptions,
-    accessToken: string
-  ): Promise<void> {
-    try {
-      if (activeSubscriptions.length > 0) {
-        const connectedUsers = await getConnectedUsersByTokenOrFail(
-          ProviderSource.fitbit,
-          fitbitUserId
-        );
-
-        await Promise.all(
-          connectedUsers.map(async user => {
-            if (
-              user.dataValues.id !== currentUser.dataValues.id &&
-              user.dataValues.cxId === currentUser.dataValues.cxId
-            ) {
-              await this.oauth.revokeLocal(user);
-              user = await getConnectedUserOrFail({
-                id: user.id,
-                cxId: user.cxId,
-              });
-              sendProviderDisconnected(user, [ProviderSource.fitbit]);
-            }
-          })
-        );
-        await this.deleteSubscriptions(activeSubscriptions, accessToken);
-      }
-    } catch (err) {
+    if (rejected.length > 0) {
       console.log(
-        `Failed to revoke the token of a Fitbit user. User: ${currentUser}, Error: ${err}`
+        `Failed to revoke the token of a Fitbit user. User: ${
+          currentUser.id
+        }, Error: ${JSON.stringify(rejected, null, 2)}`
       );
-      capture.error(err, {
-        extra: { context: "fitbit.revokeTokenFromExistingUsers", err, user: currentUser },
+      capture.message(`Failed to revoke the token of a Fitbit user.`, {
+        extra: { rejected, fitbitUserId, currentUser },
```

Comment:
If we get an error while deleting a subscription, we're sending two issues to Sentry, one on `deleteSubscriptions` and one here. Let's do it only once, please.

In fact, since `deleteSubscriptions` is only called here, I recommend we change it to delete only one subscription there and have error handling code only here.

It's better to keep error handling upstream, so we can reuse downstream/smaller functions w/o worrying whether they are handling errors or not. We just need to consider situations where:
- we need to handle errors downstream so we can execute some additional logic when it happens;
- we need to surface information that's only available downstream (in this cases, we can use wrap the error w/ `MetriportError`, which allows us to add additional context to it).